src/Model/Resolver/Products/DataPostProcessor/Attributes.php

```diff
@@ -330,6 +345,14 @@ public function process(
             // Create storage for future attributes
             $productAttributes[$productId] = [];
 
+            // Loads white-listed attrobutes - used in config product configuration
+            $this->optionCollection->addProductId((int)$productId);
+            $configAttributes = $this->optionCollection->getAttributesByProductId((int)$productId);
```

Comment:
VERY BAD PRACTISE! You should NEVER EVER do the loading in the loop. Can we trying loading this once for all products?

Reply:
Changed whole implementation...
Instead of using optionCollection I switched order of how data is loaded - firstly loads attributes options and then values, because attributes with options where the same attributes that where in whitelist in previous implementation... and now I can check if options are set instead of checking if attribute is in whitelisted array